src/glang/codegen/builtin_callables.py

```diff
@@ -138,6 +146,8 @@ def generate_ir(self, reg_gen: Iterator[int]) -> list[str]:
             assert isinstance(self._arg_type.definition, BoolDefinition)
             ir = self.UNSIGNED_IR
 
+        assert ir is not None
```

Comment:
Would it be cleaner if we defined these
```py
    FLOATING_POINT_IR = None
    SIGNED_IR = None
    UNSIGNED_IR = None
    USER_FACING_NAME = None
```
as abstract properties, so that we don't have to check if it's `None` or not

Reply:
Hmm, ur right. Sometimes it doesn't make sense to define them all eg. bitwise or doesn't need a floating point version... But this seems like something oop can fix

Reply:
Maybe OOP would be nicer, although I'd rather avoid it if possible